Simplify chapter 2 startup

wangrunji0408 opened the pull request:
完善了第一章并简化了第二章的部分内容。

对应的代码：https://github.com/rcore-os/rCore_tutorial/tree/src-20191210
目前全线都可以正常运行！

## 主要修改
### 1.1 工具链
- [x] 升级 Rust nightly 版本到 2019-12-08
- [x] 指出了锁定 nightly 版本的原因
- [x] 使用 rust-toolchain 文件

### 1.2 创建项目
- [x] 删除了 `--edition 2018`（这是默认参数）

### 2.1 目标三元组
- [x] 使用编译器内置的 riscv64 target，删除了自定义的 `riscv64-os.json`
- [x] 移除了 linker 相关内容，留到下一节再提
- [x] 移除了 abort 相关内容，因为并不是由 panic-strategy 引起的，留到以后再提
- [x] 提到了 riscv64 target 未安装错误及解决方法

### 2.2 编译内核镜像（原2.4）
- [x] 去掉 cargo xbuild 相关内容
- [x] 去掉 GCC 工具链，使用 cargo 自带的 LLVM 工具链
- [x] 增加了 `.cargo/config` 文件用来配置默认 target
- [x] 用 objdump 查看编译出的可执行文件

### 2.3 链接脚本（原2.2）
- [x] 改在 `.cargo/config` 中配置 linker
- [x] 修改链接起始地址为 0x80200000

### 2.4 入口点（原2.3）
- [x] 去掉建立初始页表的迷之代码
- [x] 去掉预编译 OpenSBI，使用 QEMU 内置的

### 2.5 QEMU
- [x] 添加安装 QEMU 的参考命令
- [x] 更新 Makefile

### 3.4 TrapFrame
- [x] 不再去掉C压缩指令集，而是让 sepc += 2

### 4.1 物理内存分配
- [x] 修改内核虚地址
- [x] 改进输出格式

### 5.1 页表概述
### 5.2 建立初始页表
- [ ] 将前面的魔法改为现在的魔法（从这里开始才建立初始页表）

### 8.1 编写用户程序
- [x] 使用内置 riscv64 target

### 8.3 链接用户程序
- [x] 改用编译脚本实现，解决每次更新用户镜像后必须 clean 的问题

### 9.1 文件系统
- [x] 更新 Makefile（修复 `rcore-fs-fuse` 版本号）

### build
- [x] 更新所有 commit ID（同时写了一个脚本自动填写）
- [x] 生成 HTML

wangrunji0408 committed:
update chapter1 part1

lock nightly version

wangrunji0408 committed:
update chapter1 part2

remove '--edition 2018' since it's added by default

wangrunji0408 committed:
update chapter1 part3-5

wangrunji0408 committed:
update chapter2 part1

use builtin riscv64 target. add cargo config file.

chyyuu commented:
同意提交！
不过我验证了最后部分（fs），有编译错误，需要进一步修改。

chyyuu commented:
同意提交！
不过我验证了最后部分（fs），有编译错误，需要进一步修改。

chyyuu reviewed (approved):
同意提交！
不过我验证了最后部分（fs），有编译错误，需要进一步修改。

wangrunji0408 committed:
update chapter1 part1

use rust-toolchain file to specify nightly version

wangrunji0408 committed:
update chapter2 part4: build kernel image

move 'cargo build' from part1 to part4.
use cargo-binutils to replace GNU toolchain.
add disassemble.

wangrunji0408 committed:
reorder chapter2

advance part4(compile) before part2

wangrunji0408 committed:
update chapter2 part3: linker script

use cargo config. fix base address.

wangrunji0408 committed:
update chapter2 part4: rewrite entry

remove magic code

wangrunji0408 committed:
update chapter2 part5: run on QEMU

add command to install QEMU on Linux or macOS.
update Makefile.

wangrunji0408 committed:
update chapter2 part6-8

wangrunji0408 committed:
update chapter3: interrupt

sepc += 2

wangrunji0408 committed:
update chapter4 part1: frame allocator

update format

wangrunji0408 committed:
update chapter4 part2: heap allocator

move global allocator to memory mod.

wangrunji0408 committed:
update chapter8 part3: link user image

use build script to ensure linking the latest user image

wangrunji0408 committed:
update chapter8 part1: user program

use builtin target.
update Makefile for user.

wangrunji0408 committed:
Merge remote-tracking branch 'origin/master' into simplify

# Conflicts:
#	SUMMARY.md
#	chapter1/part1.md
#	chapter1/part2.md
#	chapter1/part3.md
#	chapter2/part2.md
#	chapter2/part3.md
#	chapter2/part4.md
#	chapter2/part7.md
#	chapter3/part4.md
#	chapter8/part1.md
#	chapter9/part1.md

wangrunji0408 committed:
fix ```asm -> riscv

wangrunji0408 committed:
update commit ID and add filling script

wangrunji0408 committed:
fix rcore-fs-fuse commit version

ref: https://github.com/rcore-os/rCore_tutorial/pull/1

wangrunji0408 committed:
build gitbook
